lua/entities/gmod_wire_egp/lib/egplib/usefulfunctions.lua

```diff
@@ -258,6 +258,172 @@ function EGP:DrawLine( x, y, x2, y2, size )
 	end
 end
 
+function EGP:DrawPath( vertices, size, closed )
+	if size < 1 then size = 1 end
+	local num = #vertices
+
+	if size == 1 then
+		if closed then
+			vertices[num+1] = vertices[1]
+			num = num+1
+		end
+		local last = vertices[1]
+		for i=2, num do
+			local v = vertices[i]
+			surface.DrawLine( last.x, last.y, v.x, v.y )
+			last = v
+		end
+	else
+		if closed then
+			-- pad array to simplify calculation
+			vertices[num+1] = vertices[1]
+			vertices[num+2] = vertices[2]
+			num = num+1
+		end
+
+		local x1 = vertices[1].x
+		local y1 = vertices[1].y
+
+		for i=2, num do
+			local x2 = vertices[i].x
+			local y2 = vertices[i].y
+
+			local tempx = x2
+			local tempy = y2
+
+			local len = math.sqrt( (x2-x1) ^ 2 + (y2-y1) ^ 2 )
+			local dir = {x=(x2-x1)/len, y=(y2-y1)/len}
+
+			if closed and i==2 then
+				-- Offset start point to match with closing line
+				local x0 = vertices[num-1].x
+				local y0 = vertices[num-1].y
+				local len0 = math.sqrt( (x1-x0) ^ 2 + (y1-y0) ^ 2 )
+				local dir0 = {x=(x1-x0)/len0, y=(y1-y0)/len0}
+
+				local ang = math.abs(math.acos(dir0.x*dir.x + dir0.y*dir.y)) -- angle between previous line segment and this one
+				if math.deg(ang)<=90 then
+					local shift = size/2 * math.tan(ang/2)
+					x1 = x1 - shift * dir.x
+					y1 = y1 - shift * dir.y
+					len = len+shift
+				end
+			end
+
+			if closed or i<num then
+				-- Offset end point to match with next line
+				local x3 = vertices[i+1].x
+				local y3 = vertices[i+1].y
+				local len2 = math.sqrt( (x3-x2) ^ 2 + (y3-y2) ^ 2 )
+				local dir2 = {x=(x3-x2)/len2, y=(y3-y2)/len2}
+
+				local ang = math.abs(math.acos(dir.x*dir2.x + dir.y*dir2.y)) -- angle between current line segment and the next one
+				if math.deg(ang)<=90 then
+					local shift = size/2 * math.tan(ang/2)
+					x2 = x2 + shift * dir.x
+					y2 = y2 + shift * dir.y
+					len = len+shift
+
+					tempx = tempx - shift * dir2.x
+					tempy = tempy - shift * dir2.y
+				end
+			end
+			-- Calculate position
+			local xc = (x1 + x2) / 2
+			local yc = (y1 + y2) / 2
+
+			-- Calculate angle (Thanks to Fizyk)
+			local angle = math.deg(math.atan2(y1-y2,x2-x1))
+
+			-- if the rectangle's less than a pixel wide, nothing will get drawn.
+			if len < 1 then len = 1 end
+
+			surface.DrawTexturedRectRotated( math.Round(xc), math.Round(yc), math.ceil(len), size, angle )
+
+			x1 = tempx
+			y1 = tempy
+		end
+	end
+end
+
+function EGP:DrawPathPoly( vertices, size, closed )
+	if size < 1 then size = 1 end
+	local num = #vertices
+
+
+	if size == 1 then -- size 1 => just normal lines
+		if closed then -- if closed shape, duplicate first point to end
+			vertices[num+1] = vertices[1]
```

Comment:
Is modifying the `vertices` array safe? It looks like the passed in array isn't always created on the spot, therefore it'll continue to grow larger.